runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -275,6 +276,25 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * Check if the is_federated flag is changed or not, which should only happen during rolling
+     * upgrade. We should update the registry table when the flag is changed even if there is
+     * already an entry of the current table.
+     *
+     * @param oldRecord    - Previous record of the current table if it was opened before.
+     * @param newMetadata  - Metadata of the table that is being registered.
+     * @return true if oldRecord is null or the flag is changed. Otherwise, return false.
+     */
+    private boolean federatedFlagChanged(CorfuRecord<TableDescriptors, TableMetadata> oldRecord,
+                                         TableMetadata.Builder newMetadata) {
+        if (oldRecord == null) {
+            return true;
+        }
+        return oldRecord.getMetadata().getTableOptions().getIsFederated() !=
```

Comment:
we should also check the version of the protobuf as done in tryUpdateTableSchemas().  The flag change is only allowed if the version has changed.

Reply:
I think one possible case is that, if the schema change is from above layers other than corfu side, there won't be a change of the `CorfuSourceCodeVersion`, then our version check here won't be able to catch the changes.

Reply:
Actually this brings up a good question - how do we end up in a scenario where the protobuf has not changed but the TableOptions have?
Isn't `is_federated` a field specified in the protobuf? That means nobody can change that field without modifying the protobuf anyway right? And we do capture changes to the protobuf and already update the TableRegistry if protobuf file has changed?

Reply:
Yeah, fields like `is_federated` flag have to be changed by modifying protobuf files, but modifying protobuf files might not trigger a change in `CorfuSourceCodeVersion` (for example, a protobuf file changed from NSX side). As our version check relies on corfu commit hash, it cannot capture and update the registry table in that case.

Reply:
During internal testing, it is possible that different product(non-corfu) versions use the same corfu version with different values for the flag.  But it will not be the case in a real deployment.  Also, tryUpdateTableSchema() currently does not support changing the schema in the same version.  I think we should keep the same behavior.

@hisundar , the schema of the protobuf has not changed here.  So we will not capture it as per my understanding.